ownCloud/Client/Actions/EditDocumentViewController.swift

```diff
@@ -95,26 +96,45 @@ class EditDocumentViewController: QLPreviewController, Themeable {
 		}
 	}
 
-	override func viewDidAppear(_ animated: Bool) {
-		super.viewDidAppear(animated)
-		enableEditingMode()
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+		timer = nil
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		let queue = DispatchQueue(label: "com.owncloud.edit-document-queue")
+		timer = DispatchSource.makeTimerSource(queue: queue)
+		timer!.schedule(deadline: .now(), repeating: .seconds(1))
+		timer!.setEventHandler { [weak self] in
+			OnMainThread {
+				if (UIDevice.current.isIpad && self?.navigationItem.rightBarButtonItems?.count ?? 0 > 1) || (!(self?.navigationController?.isToolbarHidden ?? false)) {
```

Comment:
@hosy may be split this long if condition in few guards?

Reply:
Could not split this in guard statements, because one statement is related for the iPad and the other one for the iPhone. If I add guard statements for each device, the second guard statement could not be reached.

If I am wrong, please append a proposal to the CR.